src/util.h

```diff
@@ -29,6 +29,16 @@
 #include <boost/signals2/signal.hpp>
 #include <boost/thread/exceptions.hpp>
 
+#ifdef DEBUG_ASSERTION
+/// If DEBUG_ASSERTION is enabled this asserts when the predicate is false.
+//  If DEBUG_ASSERTION is disabled and the predicate is false, it executes the execInRelease statements.
+//  Typically, the programmer will error out -- return false, raise an exception, etc in the execInRelease code.
+#define DbgAssert(pred, execInRelease) do { assert(pred); } while(0)
+#else
+#define DbgStringify(x) #x
```

Comment:
Sorry I was working on a better printout and left it in.  Its part of a preprocessor trick to convert integer literals into a string.  I have now added the better prints and a unit test.  An example of the output is:
2017-03-23 02:23:57 test/util_tests.cpp(61): Debug Assertion failed: "0"